src/s3/client/listen_bucket_notification.rs

```diff
@@ -0,0 +1,117 @@
+            impl Stream<Item = Result<NotificationRecords, Error>>,
+        ),
+        Error,
+    > {
+        if self.base_url.is_aws_host() {
+            return Err(Error::UnsupportedApi(String::from(
+                "ListenBucketNotification",
+            )));
+        }
+
+        let region = self
+            .get_region(&args.bucket, args.region.as_deref())
+            .await?;
+
+        let mut headers = Multimap::new();
+        if let Some(v) = &args.extra_headers {
+            merge(&mut headers, v);
+        }
+
+        let mut query_params = Multimap::new();
+        if let Some(v) = &args.extra_query_params {
+            merge(&mut query_params, v);
+        }
+        if let Some(v) = args.prefix {
+            query_params.insert(String::from("prefix"), v.to_string());
+        }
+        if let Some(v) = args.suffix {
+            query_params.insert(String::from("suffix"), v.to_string());
+        }
+        if let Some(v) = &args.events {
+            for e in v.iter() {
+                query_params.insert(String::from("events"), e.to_string());
+            }
+        } else {
+            query_params.insert(String::from("events"), String::from("s3:ObjectCreated:*"));
+            query_params.insert(String::from("events"), String::from("s3:ObjectRemoved:*"));
+            query_params.insert(String::from("events"), String::from("s3:ObjectAccessed:*"));
+        }
+
+        let resp = self
+            .execute(
+                Method::GET,
+                &region,
+                &mut headers,
+                &query_params,
+                Some(&args.bucket),
+                None,
+                None,
+            )
+            .await?;
+
+        let header_map = resp.headers().clone();
+
+        let body_stream = resp.bytes_stream();
+        let body_stream = body_stream
+            .map(|r| r.map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err)));
+        let stream_reader = StreamReader::new(body_stream);
+
+        let record_stream = Box::pin(stream::unfold(
+            stream_reader,
+            move |mut reader| async move {
+                let mut line = String::new();
+                loop {
+                    match reader.read_line(&mut line).await {
+                        Ok(n) => {
+                            if n == 0 {
+                                return None;
+                            }
+                            let s = line.trim();
+                            if s.is_empty() {
+                                continue;
+                            }
+                            let records_res: Result<NotificationRecords, Error> =
+                                serde_json::from_str(&s).map_err(|e| e.into());
+                            return Some((records_res, reader));
+                        }
+                        Err(e) => return Some((Err(e.into()), reader)),
+                    }
+                }
+            },
+        ));
```

Comment:
Found a simpler implementation from a library for parsing lines from the http response, added as a new commit.